### PR TITLE
Record: Print statistics on exit

### PR DESCRIPTION
**Pull request by jbeyerstedt**
Print a list of recevied topics with average message rate and size.

Output looks like this:
```
Message Statistics by Topic:
hmi/tx/ble/glosa/spat    :   808 msg,     56/    56/    56 byte,  135/ 254/ 373 ms delta (min/avg/max)
hmi/tx/ble/own_pos       :   808 msg,     83/    83/    83 byte,  136/ 254/ 373 ms delta (min/avg/max)
hmi/tx/json/cam          :  1063 msg,    225/   252/   272 byte,    0/ 193/ 551 ms delta (min/avg/max)
hmi/tx/proto/glosa       :   808 msg,     54/    54/    54 byte,  135/ 254/ 373 ms delta (min/avg/max)
hmi/tx/proto/simplecam   :  1063 msg,     84/   153/   224 byte,    0/ 193/ 551 ms delta (min/avg/max)
v2x/rx/cam               :  1063 msg,    167/   246/   327 byte,    0/ 193/ 583 ms delta (min/avg/max)
v2x/rx/obu_gnss          :   808 msg,   3481/  3662/  3764 byte,  132/ 254/ 385 ms delta (min/avg/max)
```

### CODE REVIEW

**Comment by jbeyerstedt** [comment]
@steffens-cit Can you test the build output on you M1 Mac? Either from the CI or from the preliminary release.

**Review by steffens-cit — CHANGES_REQUESTED** [review]
Discuss max/min values. Rest is working on M1.

**Comment by jbeyerstedt** [comment]
How about:
```
v2x/rx/cam               :     2 msg,    255/   254/   255 byte (min/avg/max),  605/ 611/ 618 ms delta (min/avg/max)
v2x/rx/mapem             :    22 msg,    653/  1023/  1339 byte (min/avg/max),   16/  69/ 199 ms delta (min/avg/max)
v2x/rx/obu_gnss          :     6 msg,   3566/  3576/  3583 byte (min/avg/max),  237/ 277/ 308 ms delta (min/avg/max)
v2x/rx/spatem            :    67 msg,    435/   683/   930 byte (min/avg/max),    7/   0/ 109 ms delta (min/avg/max)
```

**Comment by steffens-cit** [comment]
Bildschirmfoto 2022-08-09 um 11 00 55

Perfect!

**Comment by jbeyerstedt** [comment]
But the average of 106 message with min size of 56 byte and max size of 56 bytes can't be zero, right?

**Comment by steffens-cit** [comment]
Also is not possible that the average value is lower than the min value (obu_gnss). The fuck?

File: mqtt-record/main.go [hunk 1]
```diff
@@ -78,6 +98,46 @@ var message_handler mqtt.MessageHandler = func(client mqtt.Client, msg mqtt.Mess
...
+
+	// calculate message statistics (omitting first message)
+	stats, exists := msgStats[topic]
+	if !exists {
+		var initStats = MsgStats{}
+		initStats.NumMsgs = 0
+		initStats.AvgMsgSizeByte = 0
+		initStats.AvgTimeDiffMillis = 0
+		initStats.LastMsgMillis = t
+
```
Comment (steffens-cit) [inline mqtt-record/main.go:110]: The initial values for MinTimeDiffMillis etc. are missing. Does this causes the error on output?

Comment (jbeyerstedt) [reply mqtt-record/main.go:110]: I'm refactoring that code currently. But somehow the average still goes to zero after some time. Min and max are fine.

Comment (jbeyerstedt) [reply mqtt-record/main.go:110]: Probably integer divisions where floating point should have been.

**Comment by jbeyerstedt** [comment]
> Also is not possible that the average value is lower than the min value (obu_gnss). The fuck?

The averaging was doing integer divisions instead of floating point, so everything got rounded down every iteration. Output is stable (reaching the right value) now.
```
Message Statistics by Topic:
hmi/tx/ble/glosa/spat    :   808 msg,     56/    56/    56 byte (min/avg/max),  135/ 254/ 373 ms delta (min/avg/max)
hmi/tx/ble/own_pos       :   808 msg,     83/    83/    83 byte (min/avg/max),  136/ 254/ 373 ms delta (min/avg/max)
hmi/tx/json/cam          :  1063 msg,    225/   252/   272 byte (min/avg/max),    0/ 193/ 551 ms delta (min/avg/max)
hmi/tx/proto/glosa       :   808 msg,     54/    54/    54 byte (min/avg/max),  135/ 254/ 373 ms delta (min/avg/max)
hmi/tx/proto/simplecam   :  1063 msg,     84/   153/   224 byte (min/avg/max),    0/ 193/ 551 ms delta (min/avg/max)
v2x/rx/cam               :  1063 msg,    167/   246/   327 byte (min/avg/max),    0/ 193/ 583 ms delta (min/avg/max)
v2x/rx/obu_gnss          :   808 msg,   3481/  3662/  3764 byte (min/avg/max),  132/ 254/ 385 ms delta (min/avg/max)
```

I changed the CLI params a bit again. You'll get the statistics summary every time now (even in verbosity zero mode). But you can also request to get the statistics every 5 seconds with `-s`.

**Review by steffens-cit — APPROVED** [review]

**Pull request merged**